samples/src/main/java/ecom/EcomClient.java

```diff
@@ -0,0 +1,163 @@
+        String paymentReference = generatePaymentReference();
+        String orderId = generateOrderId();
+        String currency = CurrencyEnum.HKD.name();
+        return sendPayment(paymentReference, amount, currency, orderId);
+    }
+
+    private PaymentInitiationResponse sendPayment(String paymentReference, long amount, String currency, String orderId) {
+        LOG.info("Initiating payment for amount: {}", amount);
+
+        PaymentInitiationRequest paymentInitiationRequest = PaymentInitiationRequest.newBuilder()
+                .setStoreId(storeId.toString())
+                .setPaymentReference(paymentReference)
+                .setAmount(amount)
+                .setCurrency(currency)
+                .setOrderId(orderId)
+                .setReturnUrl("returnUrl")
+                .setExpiry(PaymentInitiationRequest.ExpirySettings.newBuilder()
+                        .setShowTimer(true)
+                        .setExpiringSeconds(1800)
+                        .build()
+                ).build();
+
+        PaymentInitiationResponse response = paymentClient.sendOnlinePayment(paymentInitiationRequest);
+
+        LOG.info("Sent Online Payment - Payment Id: {}, Payment URL: {}", response.getResponse().getPaymentId(), response.getResponse().getPaymentUrl());
+        return response;
+    }
+
+    void requestRefund(String paymentId, long amount) {
+        LOG.info("Request refund for amount: {}", amount);
+
+        RefundRequest refundRequest = RefundRequest.newBuilder()
+                .setStoreId(storeId.toString())
+                .setPaymentId(paymentId)
+                .setAmount(String.valueOf(amount))
+                .build();
+
+        RefundResponse response = paymentClient.requestOnlineRefund(refundRequest);
+
+        LOG.info("Requested Online Refund - Response: {}", response);
+    }
+
+    public List<PaymentDetails> getPayments() {
+        LOG.info("Get payments");
+
+        GetPaymentsRequest getPaymentsRequest = GetPaymentsRequest.newBuilder()
+                .setStoreId(storeId.toString())
+                .setPageCursor(PageCursor.newBuilder().setPageSize(1).build())
+                .build();
+
+        List<PaymentDetails> response = paymentClient.getPayments(getPaymentsRequest);
+
+        LOG.info("Payments: {}", response.stream()
+                .map(p -> "PaymentId: " + p.getPaymentId() + ", OrderId: " + p.getOrderId())
+                .collect(Collectors.joining(", ")));
+        return response;
+    }
+
+    public static void main(String[] args) throws InterruptedException {
+        long amountInPence = 100;
+
+        Optional<PaymentDetails> payment;
+        PaymentInitiationResponse paymentResponse = ecomClient.sendPayment(amountInPence);
+
+        // Wait for payment to be complete before refunding it
+        do {
+            LOG.info("Waiting for online payment to complete");
+            Thread.sleep(5000);
+            payment = ecomClient
+                    .getPayments()
```

Comment:
Could the API change to include a single paymentId filter ?
At the moment it will return _all_ payments for a store ?

Reply:
GetPaymentsRequest only takes a store Id and a pageCursor currently. But at least when we do the call we ask for only one page of the results

Reply:
In kp-wombat we have

```LIMIT ${criteria.pageSize} OFFSET ${criteria.page * criteria.pageSize}```

Reply:
`pageSize = 1` means return one record, and `page = 0` means return payments from the beginning of the data set.

Reply:
Exactly yes